modules/indexer/code/search.go

```diff
@@ -125,12 +127,12 @@ func searchResult(result *internal.SearchResult, startIndex, endIndex int) (*Res
 
 // PerformSearch perform a search on a repository
 // if isFuzzy is true set the Damerau-Levenshtein distance from 0 to 2
-func PerformSearch(ctx context.Context, repoIDs []int64, language, keyword string, page, pageSize int, isFuzzy bool) (int, []*Result, []*internal.SearchResultLanguages, error) {
-	if len(keyword) == 0 {
+func PerformSearch(ctx context.Context, opts *SearchOptions) (int, []*Result, []*SearchResultLanguages, error) {
+	if opts == nil || len(opts.Keyword) == 0 {
```

Comment:
I do not think `opts == nil` makes sense, how could it be nil? If "nil" opts means a but, it should be handled in development stage.

Reply:
It should not be nil ... but better not panic

Reply:
>  It should not be nil ... but better not panic

If it shouldn't be nil, why not report the errors in development stage?

I really dislike hiding errors. It makes it more difficult to debug some bugs.

Reply:
NO we should not panic if we can arange it. To hope for stacktraces to get reported as "_lintin_" methode is wrong !!!

But what we can do is create a proper error to check against

Reply:
NO, why it would panic if there is no bug in code?

Golang library itself panics a lot if something is totally wrong.

Reply:
I also don't think we should check `opts == nil` here. It should not be nil if our code is right. Otherwise, you need to check another thousand places.

Reply:
Let me check what comsumers this func has ... and also introduce a proper error instead of a ignore on keyword==""

Reply:
Hmm ... I only mean `opts == nil` check doesn't make sense.

`if keyword == ""` looks good to me, I do not think it's worth to introduce to many "errors". 

Golang has a very bad error system, TBH I have been tired of writing a lot of `if err != nil { ctx.ServerError }`

Reply:
Do you wana create a pull?

Reply:
Since it has been merged, I think it could be left there as-is, no good no harm.